### DISCUSSION TITLE
APIRouter endpoints are not changing with given FastApi root_path

### DISCUSSION THREAD

**Discussion by lcsvcn** [fastapi/fastapi#8455] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def create_app() -> FastAPI:
    docs_url = config('DOCS_URL', "/docs")
    redoc_url = config('REDOC_URL', "/redoc")
    openapi_url = config("OPEN_API_URL", "/openapi.json")
    root_path = config("ROOT_PATH", "")

    app = FastAPI(
        docs_url=docs_url,
        redoc_url=redoc_url,
        openapi_url=openapi_url,
        root_path=root_path
    )

    # Register Api Routes
    register_routes(app)

    return app


def register_routes(app: FastAPI):
    app.include_router(
        route,
        prefix="/api/route",
        tags=['route']
    )

app = create_app()


route = APIRouter()


@test.get(
    "/status",
)
def get_status():
    return True

if __name__ == "__main__":
    uvicorn.run("middleware.application:app", reload=True)
```

### Description

Passing a **ROOT_PATH** enviroment will change where the **/docs** and **/redoc** and **/openapi.json** are deployed, but APIRouter ignores root_path completely.

Making documentation not work on docs/

Example:

ROOT_PATH=testing/

/docs will be: 

> myapacheserver.com/testing/docs/

/redoc will be: 

> myapacheserver.com/testing/redoc/

openapi.json will be:

> myapacheserver.com/testing/openapi.json 

But when you go to swagger and try do do a GET:
```

curl -X 'GET' \
  'api/route/status' \
  -H 'accept: application/json
  Response:
  
```

> detail: { Not Found }

  
Since api is at:s at 

```
curl -X 'GET' \
  'testing/api/route/status' \
  -H 'accept: application/json
```

### Wanted Solution

  What I think that should happen:
  
  APIRouter should be able to receive an ROOT_PATH or use the same ROOT_PATH as the open_api.json, docs and redoc
  
  
  Rn I tried everything to make swagger api to work from the documention, but without success.
  
  Because root_path is only applied to openapi.json, /docs and /redoc and there is no way to change APIroute root_path

### Wanted Code

```python
def create_app() -> FastAPI:
    docs_url = config('DOCS_URL', "/docs")
    redoc_url = config('REDOC_URL', "/redoc")
    openapi_url = config("OPEN_API_URL", "/openapi.json")
    root_path = config("ROOT_PATH", "")

    app = FastAPI(
        docs_url=docs_url,
        redoc_url=redoc_url,
        openapi_url=openapi_url,
        root_path=root_path
    )

    # Register Api Routes
    register_routes(app)

    return app


def register_routes(app: FastAPI):
    app.include_router(
        route,
        prefix="/api/route",
        tags=['route']
    )

app = create_app()


route = APIRouter()


@test.get(
    "/status",
)
def get_status():
    return True

if __name__ == "__main__":
    uvicorn.run("middleware.application:app", reload=True)
```

### Alternatives

Tried to provide:

route = APIRouter(prefix=root_path)

This result in sth like: root_path/api/route/root_path/status (twice the root path)
Also tried:

app.include_router(
        route,
        prefix=f"{root_path}/api/route",
        tags=['route']
    )
    
This result in sth like: root_path/root_path/api/route/status (twice the root path)

### Operating System

Linux

### Operating System Details

Fedora 35

### FastAPI Version

0.25.0

### Python Version

python 3.9.10

### Additional Context

Using Apache to expose the API / documentation

**Comment by PierroD** [comment]
If I understood correctly, you want to get an access to your root using the root_path parameters.
But the root_path has been created to protect your documentation to be read by anyone, cf doc (https://fastapi.tiangolo.com/advanced/behind-a-proxy/) 

If you need the same URL for your API and your documentation, you can simply  add an "prefix" to your Routers like you did in your example (APIRouter(prefix=root_path)).

Last idea, if your URL got root_path twice, you can simply delete the one you wrote in the APIRouter.
